src/v/config/configuration.cc

```diff
@@ -1766,6 +1766,19 @@ configuration::configuration()
       "exception is thrown instead.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       true)
+  , sampled_memory_profile(
+      *this,
+      "memory_enable_memory_sampling",
+      "If true, memory allocations will be sampled and tracked. A sampled live "
+      "set of allocations can then be retrieved from the Admin API. "
+      "Additionally, we will periodically log the top-n allocation sites",
+      {// We could use a watcher and change this at runtime. However it's not
+       // entirely clear what that would mean for the profile, i.e.: you might
+       // be missing a large set of allocations if this gets changed after
+       // start. Possibly disabling at runtime is useful though?
+       .needs_restart = needs_restart::yes,
```

Comment:
Seems like it should be posisble to make this `needs_restart::no`? This is _really_ handy both for actual escalations (where people _hate_ to disable their servers) and for testing where you can flip something on and off to see its effect.

Reply:
I have made it so now. However, as the comment was indicating turning this on/off doesn't make that much sense from the profiling perspective but for perf testing or something it might still so yeah it's done.